src/sentry/api/invite_helper.py

```diff
@@ -38,17 +39,46 @@ def get_invite_cookie(request):
 
 
 class ApiInviteHelper(object):
-    def __init__(self, instance, request, member_id, token, logger=None):
+    @classmethod
+    def fromCookieOrEmail(cls, request, organization, email, instance=None, logger=None):
+        """
+        Initializes the ApiInviteHelper by locating the pending organization
+        member via the currently set pending invite cookie, or via the passed
+        email if no cookie is currently set.
+        """
+        pending_invite = get_invite_cookie(request)
+
+        try:
+            if pending_invite is not None:
+                om = OrganizationMember.objects.get(
+                    id=pending_invite["memberId"], token=pending_invite["token"]
```

Comment:
When `pending_invite` comes from a cookie do we have any constraints on that cookie value that prevent the `memberId` being tampered with?

Reply:
Yeah the token must also match. Same requirement as the accept page URL